### DISCUSSION TITLE
Soft limit errors

### DISCUSSION THREAD

**Discussion by JJTechman68** [phil-barrett/PicoCNC#19] [Q&A] [answered]
I am getting soft limit errors on my custom machine using the PicoCNC and Gsender. (I am a Linux user so IO sender is a no go for me) I have set the bed size, the machine homes properly and the gantry moves in the correct directions when jogging. If I turn soft limits off, everything works flawlessly. As soon as I turn them on and try to run even a test run, a soft limit error pops up.

So, does anybody know if this is an issue with GrblHAL or using proximity sensors? If I can't get soft limits to function properly, I can wire in hard limits for max travel but how would that interface with the PicoCNC?

**Comment by phil-barrett** [comment, marked as answer]
Did you want soft limits?  If you have prox sensors, enable hard limits.   Here's a blog I did on [proximity sensors](https://www.grbl.org/single-post/using-proximity-sensors-with-the-picocnc).

**Reply by JJTechman68** [reply]
Thanks Phil. Turning off the soft limits worked.
